### PR TITLE
Fix #1475: network config not being saved by CLI && AZ

### LINKED ISSUE #1475: network configuration isn't properly being saved

### PR DESCRIPTION
**Pull request by subtly**

### CODE REVIEW

File: libp2p/Host.cpp [hunk 1]
```diff
@@ -692,27 +689,23 @@ bytes Host::saveNetwork() const
...
-				network << p.peerEndpoint().port() << p.id << 0
-					<< chrono::duration_cast<chrono::seconds>(p.m_lastConnected.time_since_epoch()).count()
-					<< chrono::duration_cast<chrono::seconds>(p.m_lastAttempted.time_since_epoch()).count()
-					<< p.m_failedAttempts << (unsigned)p.m_lastDisconnect << p.m_score << p.m_rating;
-				count++;
-			}
+			// todo: e2e ipv6 support
+			if (endpoint.address().is_v4())
+				network << endpoint.address().to_v4().to_bytes();
+				continue;
```
Comment (gavofyork) [inline libp2p/Host.cpp]: just a tab issue or did you mean to parenthesise?

File: libp2p/Host.cpp [hunk 2]
```diff
@@ -731,10 +724,13 @@ bytes Host::saveNetwork() const
 			count++;
 		}
 	}
+	// else: TODO: use previous configuration if available
 
 	RLPStream ret(3);
 	ret << dev::p2p::c_protocolVersion << m_alias.secret();
-	ret.appendList(count).appendRaw(network.out(), count);
+	ret.appendList(count);
+	if (count)
```
Comment (gavofyork) [inline libp2p/Host.cpp]: `!!count`

File: libp2p/Host.cpp (hunk 1 above)
Comment (subtly) [inline libp2p/Host.cpp]: Nope. Bug.

**Comment by subtly** [comment]
Fixed. The unit test didn't work out because 'locahost' won't be saved as a peer unless it's required. Will require additional unit test framework and be revisited when ipv6 is added.

**Pull request merged**